src/Nerdbank.Streams.Tests/NestedStreamTests.cs

```diff
@@ -47,15 +47,21 @@ public void Slice_InputValidation()
     [Fact]
     public void Length()
     {
-        Assert.Throws<NotSupportedException>(() => this.stream.Length);
+        Assert.Equal(10, this.stream.Length);
         this.stream.Dispose();
         Assert.Throws<ObjectDisposedException>(() => this.stream.Length);
     }
 
     [Fact]
     public void Position()
     {
-        Assert.Throws<NotSupportedException>(() => this.stream.Position);
+        byte[] buffer = new byte[DefaultNestedLength];
+
+        Assert.Equal(0, this.stream.Position);
+        this.stream.Read(buffer, 0, 5);
+        Assert.Equal(5, this.stream.Position);
+        this.stream.Read(buffer, 10, 0);
```

Comment:
Why is this last Read here? It reads 0-bytes, starting at a non-existent index into the buffer.